src/components/PopupWithForm.js

```diff
@@ -18,11 +22,15 @@ export default class PopupWithForm extends Popup {
 
   setEventListeners() {
     this._popupForm.addEventListener("submit", (event) => {
-      event.preventDefault();
-      //this._handleFormSubmit(({} = this._getInputValues()));
-      this._handleFormSubmit(this._getInputValues());//slight tweak of the above, seems to work and looks cleaner.
+      event.preventDefault();      
+      this._handleFormSubmit(this._getInputValues());
       this._popupForm.reset();
```

Comment:
` this._popupForm.reset();`

You need to **reset** the form only  in block `then` in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.